lib/client/client.go

```diff
@@ -108,20 +111,45 @@ func (c *Client) EnsureInitialized() error {
 		return err
 	}
 
-	// Ensure a cluster GUID is set.  This is required for all datastore
-	// types.
-	kv := &model.KVPair{
-		Key:   model.GlobalConfigKey{Name: "ClusterGUID"},
-		Value: hex.EncodeToString(uuid.NewV4().Bytes()),
+	// Ensure a number of global Felix config values are configured:
+	// -  GUID (ClusterGUID)
+	// -  IPIP (IpInIpEnabled)
+	// -  Felix reporting interval (ReportingIntervalSecs)
+	// -  Log severity for screen (LogSeverityScreen)
+	// -  Log severity for file (LogSeverityFile)
+	// -  Log file path (LogFilePath)
+	guid := hex.EncodeToString(uuid.NewV4().Bytes())
+	if err := c.ensureGlobalFelixConfig("ClusterGUID", guid); err != nil {
+		return err
+	} else if err = c.ensureGlobalFelixConfig("ReportingIntervalSecs", "0"); err != nil {
+		return err
+	} else if err = c.ensureGlobalFelixConfig("IpInIpEnabled", GlobalDefaultIPIP); err != nil {
+		return err
+	} else if err = c.ensureGlobalFelixConfig("LogSeverityScreen", GlobalDefaultLogLevel); err != nil {
```

Comment:
Really I feel that `EnsureInitialized` should not be a place where we throw a lot of defaults for things.  It's meant to make sure that the datastore is initialized to a minimal level such that things won't break (e.g the right etcd directories, or k8s thirdpartyresources exist).

If calico/node wants certain defaults, then it should configure them itself.